tests/system/test_cluster/test_agent_files_deletion/test_agent_files_deletion.py

```diff
@@ -54,14 +54,14 @@ def test_agent_files_deletion():
         host_manager.clear_file(host=hosts, file_path=os.path.join(WAZUH_LOGS_PATH, 'cluster.log'))
         host_manager.control_service(host=hosts, service='wazuh', state="restarted")
 
-    # Check if the agent is connected and reporting
-    agent_healthcheck()
-
     # Get the current ID and name of the agent that is reporting to worker_host.
     master_token = host_manager.get_api_token(master_host)
     response = host_manager.make_api_call(host=master_host, method='GET', token=master_token,
                                           endpoint=f'/agents?select=id,name&q=manager={worker_host}')
 
+    # Check if the agent is connected and reporting
+    agent_healthcheck(master_token)
+
```

Comment:
Run this healthcheck right after getting the token:

```suggestion
    # Check if the agent is connected and reporting
    master_token = host_manager.get_api_token(master_host)
    agent_healthcheck(master_token)
    response = host_manager.make_api_call(host=master_host, method='GET', token=master_token,
                                          endpoint=f'/agents?select=id,name&q=manager={worker_host}')

```